gostream/ffmpeg/avframe.go

```diff
@@ -34,16 +27,24 @@ type Frame C.struct_AVFrame
 // @note this only allocates the AVFrame itself, not the data buffers. Those
 // must be allocated through other means, e.g. with av_frame_get_buffer() or
 // manually.
-func FrameAlloc() *Frame {
-	return (*Frame)(unsafe.Pointer(C.av_frame_alloc()))
+func FrameAlloc() (Frame, error) {
+	p := C.av_frame_alloc()
+	if p == nil {
+		return Frame{}, errors.New("cannot allocate frame")
+	}
+
+	return Frame{
```

Comment:
[q] Is there any performance downside to allocating from to two pointer refs here?

Reply:
Not that I can see.